lib/discovery.js

```diff
@@ -406,18 +408,12 @@ function mixinDiscovery(MySQL, mysql) {
 
   MySQL.prototype.buildPropertyType = function(columnDefinition) {
     var mysqlType = columnDefinition.dataType;
+    var columnType = columnDefinition.columnType;
     var dataLength = columnDefinition.dataLength;
 
     var type = mysqlType.toUpperCase();
     switch (type) {
       case 'CHAR':
-        if (dataLength === 1) {
-          // Treat char(1) as boolean
-          return 'Boolean';
-        } else {
-          return 'String';
-        }
-        break;
```

Comment:
@raymondfeng if the current system interpret these characters as correct boolean values, then I can understand that it is reasonable to have it mapped as a boolean; Though I wonder if it would be possible to add a switch to change the behavior?